mwdb/model/file.py

```diff
@@ -81,30 +91,81 @@ def get_or_create(cls, file, parent=None, metakeys=None, share_with=None):
                     app_config.mwdb.s3_storage_secret_key,
                     app_config.mwdb.s3_storage_region_name,
                     app_config.mwdb.s3_storage_secure,
-                ).put_object(app_config.mwdb.s3_storage_bucket_name, file_obj._calculate_path(), file, file_size)
+                ).put_object(
+                    app_config.mwdb.s3_storage_bucket_name,
+                    file_obj._calculate_path(),
+                    file.stream,
+                    file_size
+                )
             else:
                 file.save(file_obj._calculate_path())
 
+        file_obj.upload_stream = file.stream
         return file_obj, is_new
 
     def _calculate_path(self):
-        # upload_path must not be None
-        upload_path = app_config.mwdb.uploads_folder or ""
+        if app_config.mwdb.storage_provider == StorageProviderType.DISK:
+            upload_path = app_config.mwdb.uploads_folder
```

Comment:
so uploads_folder is never none anymore?

Reply:
When uploads_folder is `/app/uploads` and `StorageProviderType.S3` is set: mwdb-core tries to upload objects with name `/app/uploads/<hash>` which throws error due to prepending slash in name.